src/api/goods.ts

```
@@ -4,13 +4,52 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => response.json())
+    .catch(error => {
+      // eslint-disable-next-line no-console
+      console.error(`Error fetching all goods: ${error}`);
+      throw error; // прокидаємо далі помилку, щоб її можна було обробити вище
+    })
```

Related: the current catch logs and rethrows, which is fine for propagation, but it doesn't cover HTTP non-OK responses (since response.json() is called unconditionally). Also consider handling/propagating JSON parse errors explicitly so callers (UI) can show a proper error message instead of failing silently.